src/sdk/namespace_fs.js

```diff
@@ -3133,6 +3157,29 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _check_version_moved recieves cur_path and checks if the version moved
+     * 1. if cur_path is in .versions/ - the function will check if the verison moved to latest
+     * 2. if cur_path is the latest version - the function will check if the version moved to .versions/
+     * @param {import('./nb').NativeFSContext} fs_context 
+     * @param {string} key 
+     * @param {string} version_id 
+     * @param {string} cur_path 
+     */
+    async _check_version_moved(fs_context, key, version_id, cur_path) {
```

Comment:
this seems to be a partial solution since the object can be moved again before checking, maybe its worth checking both paths to see if it exists at all? even though it still not complete, as we can't check both places at the same time.

Reply:
why partial? the retires cover future moves, but I guess since we don't need this function for move_to_dest_version() we can go with your approach.